libraries/AP_OpenDroneID/AP_OpenDroneID.h

```diff
@@ -0,0 +1,187 @@
+
+#define ODID_MIN_DIR         0       // Minimum direction
+#define ODID_MAX_DIR         360     // Maximum direction
+#define ODID_INV_DIR         361     // Invalid direction
+#define ODID_MIN_SPEED_H     0       // Minimum speed horizontal
+#define ODID_MAX_SPEED_H     254.25f // Maximum speed horizontal
+#define ODID_INV_SPEED_H     255     // Invalid speed horizontal
+#define ODID_MIN_SPEED_V     (-62)   // Minimum speed vertical
+#define ODID_MAX_SPEED_V     62      // Maximum speed vertical
+#define ODID_INV_SPEED_V     63      // Invalid speed vertical
+#define ODID_MIN_ALT         (-1000) // Minimum altitude
+#define ODID_MAX_ALT         31767.5f// Maximum altitude
+#define ODID_INV_ALT         ODID_MIN_ALT // Invalid altitude
+#define ODID_MAX_TIMESTAMP   (60 * 60)
+#define ODID_INV_TIMESTAMP   0xFFFF  // Invalid, No Value or Unknown Timestamp
+#define ODID_MAX_AREA_RADIUS 2550
+#define ODID_AREA_COUNT_MIN  1
+#define ODID_AREA_COUNT_MAX  65000
+
+class AP_UAVCAN;
+
+class AP_OpenDroneID
+{
+public:
+    AP_OpenDroneID();
+
+    /* Do not allow copies */
+    CLASS_NO_COPY(AP_OpenDroneID);
+
+    // parameter block
+    static const struct AP_Param::GroupInfo var_info[];
+
+    void init();
+    bool pre_arm_check(char* failmsg, uint8_t failmsg_len);
+    void update();
+    void send_dynamic_out();
+    void send_static_out();
+
+    // send pending dronecan messages
+    void dronecan_send(AP_UAVCAN *);
+
+    // handle a message from the GCS
+    void handle_msg(mavlink_channel_t chan, const mavlink_message_t &msg);
+
+    // transmit functions to manually send a static MAVLink message
+    void send_basic_id_message();
+    void send_system_message();
+    void send_self_id_message();
+    void send_operator_id_message();
+    void send_location_message();
+
+    void set_arm_status(mavlink_open_drone_id_arm_status_t &status);
+
+    // get singleton instance
+    static AP_OpenDroneID *get_singleton()
+    {
+        return _singleton;
+    }
+private:
+    static AP_OpenDroneID *_singleton;
+
+    // parameters
+    AP_Int8  _enable;
+    AP_Float _baro_accuracy;    // Vertical accuracy of the barometer when installed
+    AP_Int16 _options;
+    AP_Int8  _mav_port;
+
+    enum Options : int16_t {
+        EnableDroneCAN    = (1U << 0U),
+        EnforceArming     = (1U << 1U),
+    };
+
+    // check if an option is set
+    bool option_set(const Options option) const
+    {
+        return (uint8_t(_options.get()) & uint8_t(option)) != 0;
+    }
+
+    mavlink_channel_t _chan; // MAVLink channel that communicates with the Remote ID Transceiver
+    const mavlink_channel_t MAV_CHAN_INVALID = mavlink_channel_t(255U);
+    uint32_t _last_send_dynamic_messages_ms;
+    uint32_t _last_send_static_messages_ms;
+    uint32_t _mavlink_dynamic_period_ms; //how often are mavlink dynamic messages sent in ms. E.g. 1000 = 1 Hz
+    uint32_t _mavlink_static_period_ms; //how often are mavlink static messages sent in ms
```

Comment:
```suggestion
    const uint32_t _mavlink_dynamic_period_ms = 1000; //how often are mavlink dynamic messages sent in ms. E.g. 1000 = 1 Hz
    const uint32_t _mavlink_static_period_ms = 100; //how often are mavlink static messages sent in ms
```

uint32_t makes the maths unquestionable but is wasteful...